### PR TITLE
added go utility to run concurrent deployment/validation tests

### PR DESCRIPTION
**Pull request by dmitsh**
This change is [Reviewable](https://reviewable.io/reviews/azure/acs-engine/372)

### CODE REVIEW

File: test/acs-engine-test/acs-engine-test.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{"deployments":
```
Comment (colemickens) [inline test/acs-engine-test/acs-engine-test.json]: Can you run this through `jq` for consistency's sake?

File: test/acs-engine-test.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+###############################################################################
+
+set -e
+set -u
+set -o pipefail
+
+ROOT="${DIR}/.."
+
+# Check pre-requisites
```
Comment (colemickens) [inline test/acs-engine-test.sh]: Why not do these checks for env vars in Go?

File: test/acs-engine-test/config.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	}
+	return nil
+}
+
+func getTestConfig(fname string) (*TestConfig, error) {
+	data, err := ioutil.ReadFile(fname)
+	if err != nil {
+		return nil, err
+	}
+	config := new(TestConfig)
```
Comment (weinong) [inline test/acs-engine-test/config.go]: as I was reading the some go material, it's suggested that it's idiomatic to use var config TestConfig to initialize the variable with default values. or config := TestConfig{}.

File: test/acs-engine-test/acs-engine-test.json (hunk 1 above)
Comment (dmitsh) [inline test/acs-engine-test/acs-engine-test.json]: done

File: test/acs-engine-test.sh (hunk 2 above)
Comment (dmitsh) [inline test/acs-engine-test.sh]: done

File: test/acs-engine-test/config.go (hunk 3 above)
Comment (dmitsh) [inline test/acs-engine-test/config.go]: in this case it should be config := &TestConfig{}, as  I return pointer. Done.

**Pull request merged**